mp/src/game/client/hud_chat.cpp

```diff
@@ -53,6 +54,9 @@ void CHudChat::StartMessageMode()
     SetMouseInputEnabled(true);
     SetKeyBoardInputEnabled(true);
 
+    const auto specGUI = dynamic_cast<CMOMSpectatorGUI *>(gViewPortInterface->FindPanelByName(PANEL_SPECGUI));
```

Comment:
I think this is backwards; the chat panel shouldn't care about other particular panels.

The spec GUI on the other hand explicitly cares when chat or the map finished panels are open and have mouse input. This kind of logic should instead be there.

Reply:
Did this that way using an event.